CSLabs.Api/Services/TestProxmoxConnectionService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using CSLabs.Api.Models;
+using CSLabs.Api.Proxmox;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace CSLabs.Api.Services
+{
+    public class TestProxmoxConnectionService
+    {
+        private DefaultContext Context { get; }
+
+        private ProxmoxManager ProxmoxManager;
+        
+        public TestProxmoxConnectionService(DefaultContext context)
+        {
+            Context = context;
+        }
+        
+        public async Task<IActionResult> TestProxmoxConnection()
+        {
+            var hypervisors = await Context.Hypervisors
+                .Include(h => h.HypervisorNodes)
+                .ToListAsync();
+            var systemStatus = Context.SystemStatuses.First(); // Get current system status
+            try
+            {
+                foreach (var hypervisor in hypervisors)
+                {
+                    var nodesUp = 0;
+                    var api = ProxmoxManager.GetProxmoxApi(hypervisor.HypervisorNodes.First());
+
+                    try 
+                    {
+                        var clusterStatus = await api.GetClusterStatus();
+                        if (!clusterStatus.Quorate)
+                        {
+                            throw new NoQuorumException();
+                        }
+                    }
+                    catch (ProxmoxRequestException)
+                    {
+                        // GetClusterStatus failed, so no nodes are up 
```

Comment:
This is not true if you do not have a quorum then that means you have 50% or less of your nodes up.
https://en.wikipedia.org/wiki/Quorum_(distributed_computing)#